noir-projects/noir-protocol-circuits/crates/types/src/meta/mod.nr

```diff
@@ -24,204 +24,28 @@ comptime fn get_trait_impl_method(
-        for _ in 0..array_len {
-            // Recursively call `generate_deserialize_from_fields` for each element in the array
-            let (deserialized_field, num_consumed_in_recursion) = generate_deserialize_from_fields(
-                name,
-                element_type,
-                field_array_name,
-                consumed_counter + num_already_consumed,
-                should_unpack,
-            );
-            // We increment the consumed counter by the number of fields consumed in the recursion
-            consumed_counter += num_consumed_in_recursion;
-            // We add the deserialized field to the list of deserialized fields.
-            array_fields_list = array_fields_list.push_back(deserialized_field);
-        }
-
-        // We can construct the array from the deserialized fields
-        let array_fields = array_fields_list.join(quote {,});
-        result = quote { [ $array_fields ] };
-    } else if typ.as_str().is_some() {
-        // The field is a string and we expect each byte of the string to be represented as 1 field in the field
-        // array. So we iterate over the string length and deserialize each character as u8 in the recursive call
-        // to `generate_deserialize_from_fields`.
-        let length_type = typ.as_str().unwrap();
-        let str_len = length_type.as_constant().unwrap();
-        let mut byte_list = &[];
-
-        // Iterate over each character in the string
-        for _ in 0..str_len {
-            // Recursively call `generate_deserialize_from_fields` for each character in the string
-            let (deserialized_field, num_consumed_in_recursion) = generate_deserialize_from_fields(
-                name,
-                quote {u8}.as_type(),
-                field_array_name,
-                consumed_counter + num_already_consumed,
-                should_unpack,
-            );
-
-            // We should consume just one field in the recursion so we sanity check that
-            assert_eq(
-                num_consumed_in_recursion,
-                1,
-                "Incorrect number of fields consumed in string deserialization",
-            );
-
-            // We increment the consumed counter by 1 as we have consumed one field
-            consumed_counter += 1;
-
-            // We add the deserialized field to the list of deserialized fields.
-            // E.g. `fields[6] as u8`
-            byte_list = byte_list.push_back(deserialized_field);
-        }
-
-        // We construct the string from the deserialized fields
-        let bytes = byte_list.join(quote {,});
-        result = quote { [ $bytes ].as_str_unchecked() };
-    } else {
-        panic(
-            f"Unsupported type for serialization of argument {name} and type {typ}",
-        )
-    }
+/// Returns the number of fields required to pack a type `typ` that implements the `Packable` trait.
+pub comptime fn get_packable_len(typ: Type) -> u32 {
```

Comment:
I plan on dropping this function in a PR up the stack as it doesn't work for `Packable` implementations that work with numeric generics.